sbt-cross/src/main/scala/sbtcross/CrossProject.scala

```diff
@@ -0,0 +1,383 @@
+ *  source directories.
+ *
+ *  To fix this, you should add symlinks in the hierarchy to the shared source
+ *  directory and include them in your imported IntelliJ IDEA project (but not
+ *  in sbt). The recommended structure is as follows (for a Full CrossProject):
+ *
+ *  <pre>
+ *  bar
+ *  ├── js
+ *  │   └── src
+ *  │       ├── main
+ *  │       ├── test
+ *  │       ├── idea-shared-main/ --> project/shared/src/main
+ *  │       └── idea-shared-test/ --> project/shared/src/test
+ *  ├── jvm
+ *  │   └── src
+ *  │       ├── main
+ *  │       ├── test
+ *  │       ├── idea-shared-main/ --> project/shared/src/main
+ *  │       └── idea-shared-test/ --> project/shared/src/test
+ *  ├── native
+ *  │   └── src
+ *  │       ├── main
+ *  │       ├── test
+ *  │       ├── idea-shared-main/ --> project/shared/src/main
+ *  │       └── idea-shared-test/ --> project/shared/src/test
+ *  └── shared
+ *      └── src
+ *          ├── main
+ *          ├── test
+ *          ├── idea-shared-main/ --> project/shared/src/main
+ *          └── idea-shared-test/ --> project/shared/src/test
+ *  </pre>
+ *
+ *  Note that we do not recommend to put the symlinks in version control, since
+ *  they do not work on Windows (Git, for example, just ignores their existence
+ *  when cloning).
+ *
+ *  <h2>Pitfalls to Avoid</h2>
+ *
+ *  <h3>Altering a contained Project outside the CrossProject</h3>
+ *  Since sbt projects are immutable structures, it is important that you do not
+ *  "mutate" (i.e. create a new Project) outside of the CrossProject.
+ *
+ *  <h4>DON'T</h4>
+ *  {{{
+ *  lazy val foo = crossProject(JSPlatform, JVMPlatform, NativePlatform)
+ *
+ *  lazy val fooJVM = foo.jvm
+ *  lazy val fooJS = foo.js.settings(jsDependencies += RuntimeDOM)
+ *
+ *  // Now we have fooJS != foo.js... Dependency tracking will not work anymore.
+ *  }}}
+ *
+ *  <h4>DO</h4>
+ *  {{{
+ *  lazy val bar = crossProject(JSPlatform, JVMPlatform, NativePlatform)
+ *    .jsSettings(jsDependencies += RuntimeDOM)
+ *
+ *  lazy val barJVM = bar.jvm
+ *  lazy val barJS = bar.js
+ *  }}}
+ *
+ *  <h3>Manually setting the base of a contained Project</h3>
+ *  CrossProject puts its contained projects in a given directory structure. If
+ *  you try to work around that, things will fail (and non-existing directories
+ *  will be referenced). If you want to put your projects in a different
+ *  directory structure, you are encouraged to implement your own subclass of
+ *  [[CrossType]].
+ *
+ *  <h4>DON'T</h4>
+ *  {{{
+ *  lazy val baz = crossProject(JSPlatform, JVMPlatform, NativePlatform)
+ *    .jsConfigure(_.in(file("myJSDir")))
+ *  }}}
+ *
+ *  <h4>DO</h4>
+ *  Implement your own subclass (sub-object) of [[CrossType]].
+ *
+ */
+final class CrossProject private[sbtcross] (
+    crossType: CrossType,
+    platforms: Set[Platform],
+    val projects: Map[Platform, Project] = Map()
+) {
+
+  def aggregate(refs: CrossProject*): CrossProject = {
```

Comment:
We should emit an error if you don't have all the required platforms. We could emit a warning if we are dropping a platform